basculechecks/capabilitycheck.go

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2020 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package basculechecks
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+
+	"github.com/goph/emperror"
+)
+
+var (
+	ErrNilPrefix = errors.New("prefix regular expression cannot be nil")
+)
+
+type constCheck struct {
```

Comment:
This is a good approach, with a single interface being implemented by several concrete types.

However, since you're already encapsulating things with an interface, you could just have a plain, exported types and forgo any `New...` function:

```
type ConstCheck string

func (c ConstCheck) Authorized(....) { /* stuff */ }

type EndpointRegexCheck struct {
    PrefixToMatch *regexp.Regexp
    AcceptAllMethod string
}

func (c EndpointRegexCheck) Authorized(...) { /* stuff */ }

// etc etc for the other types
```

That all takes up less code without really giving up any information hiding.  It also gives you one less thing per type to maintain, as if you add fields to something you don't need to update a `New...` function.  You just add the field to the struct and modify `Authorized`.